### PR TITLE
GH-9562: Handle breaking changes

### LINKED ISSUE #9562: Handle deprecations for ProductListComponentService
According to #7987 and #8874 some breaking changes were introduced. Please check both PRs and either enchance schematics migrations script or update the migration docs.

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-9562

### CODE REVIEW

File: docs/migration/3_0.md [hunk 1]
```diff
@@ -77,6 +77,13 @@ The property `childRoutes` of config `cmsComponents` changed type from `Route[]`
 
 - ConverterService constructor is now using `UnifiedInjector` instead of standard `Injector` 
 
+### Property renamed in `SearchConfig` interface
+
+| Old Name | New Name 
+| --- | --- 
+| `sortCode` | `sort` 
+
+
```
Comment (bgambocjaviniar) [inline docs/migration/3_0.md:86]: is it also worth mentioning that the two interfaces (ProductListRouteParams and SearchCriteria) from the `product-list-component.service` was removed and moved to `product-list.model`?

Comment (plabadie) [reply docs/migration/3_0.md:86]: I thought we needed to write schematics and if not possible, write some doc.

Comment (bgambocjaviniar) [reply docs/migration/3_0.md:86]: I don't think it would be in any schematics but we should put it in the doc

Comment (bgambocjaviniar) [reply docs/migration/3_0.md:86]: ------ Unrelated
just going to put it here, the branch is called `featrue/xxx` and not `feature/xxx`, but it doesn't seem to be a problem with our checks (just pointing it out)

Comment (plabadie) [reply docs/migration/3_0.md:86]: Oops, yup, we did a typo on the branch name.

Comment (plabadie) [reply docs/migration/3_0.md:86]: > is it also worth mentioning that the two interfaces (ProductListRouteParams and SearchCriteria) from the `product-list-component.service` was removed and moved to `product-list.model`?

We took a second look and those 2 interfaces were not exported before they were moved.  Therefore, not a breaking change.

**Review by bgambocjaviniar — DISMISSED** [review]

File: docs/migration/3_0.md (hunk 1 above)
Comment (bgambocjaviniar) [reply docs/migration/3_0.md:86]: 🟢

**Review by bgambocjaviniar — APPROVED** [review]

**Review by znikola — APPROVED** [review]

**Pull request merged**